pallets/manta-pay/src/lib.rs

```diff
@@ -228,18 +233,24 @@ pub mod pallet {
         #[pallet::weight(T::WeightInfo::to_public())]
         #[transactional]
         pub fn to_public(origin: OriginFor<T>, post: TransferPost) -> DispatchResultWithPostInfo {
-            let origin = ensure_signed(origin)?;
+            let _origin = ensure_signed(origin)?;
             ensure!(
                 post.sources.is_empty()
                     && post.sender_posts.len() == 2
                     && post.receiver_posts.len() == 1
-                    && post.sinks.len() == 1,
+                    && post.sinks.len() == 1
+                    && post.sink_accounts.len() == 1,
                 Error::<T>::InvalidShape
             );
             for sink in post.sinks.iter() {
                 ensure!(asset_value_decode(*sink) > 0u128, Error::<T>::ZeroTransfer);
             }
-            Self::post_transaction(None, vec![], vec![origin], post)
+            Self::post_transaction(
+                None,
+                vec![],
+                vec![T::AccountId::from(post.sink_accounts[0])],
```

Comment:
This line could panic. The check above prevents a panic from being possible, but maybe add comment saying so

Reply:
Oh yeah, let's put in the release PR because it will stop the CI.
https://github.com/Manta-Network/Manta/pull/1033